### PR TITLE
MessageListItem: Star a message by clicking the star icon

### LINKED ISSUE #259: Ability to star a message
<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/64895388-ability-to-star-a-message?utm_campaign=plugin&utm_content=tracker%2F59850425&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F59850425&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by danrabbit**
Fixes #259

### CODE REVIEW

File: src/MessageList/MessageListItem.vala [hunk 1]
```diff
@@ -248,13 +257,26 @@ public class Mail.MessageListItem : Gtk.ListBoxRow {
...
             expanded = !expanded;
-            return false;
+            return Gdk.EVENT_STOP;
         });
 
         destroy.connect (() => {
             loading_cancellable.cancel ();
         });
 
+        starred_button.button_release_event.connect (() => {
```
Comment (tintou) [inline src/MessageList/MessageListItem.vala]: Why `button_release_event` instead of `clicked` ?

Comment (danrabbit) [reply src/MessageList/MessageListItem.vala]: So we can stop the event from propogating to the eventbox that collapses the mail

Comment (tintou) [reply src/MessageList/MessageListItem.vala]: That would be perfect to not change it in the future

File: src/MessageList/MessageListItem.vala [hunk 2]
```diff
@@ -248,13 +257,26 @@ public class Mail.MessageListItem : Gtk.ListBoxRow {
...
+            if (Camel.MessageFlags.FLAGGED in (int) message_info.flags) {
+                message_info.set_flags (Camel.MessageFlags.FLAGGED, 0);
+                starred_icon.icon_name = "non-starred-symbolic";
+                starred_icon.tooltip_text = _("Star message");
+            } else {
+                message_info.set_flags (Camel.MessageFlags.FLAGGED, ~0);
```
Comment (naaando) [inline src/MessageList/MessageListItem.vala]: What does ~0 means? 🤔

Comment (tintou) [reply src/MessageList/MessageListItem.vala]: It means complement all bits of 0, so it's all bits to 1.
With the `Camel.MessageFlags.FLAGGED` mask it means that only the `Camel.MessageFlags.FLAGGED` flag will be set to 1

**Pull request merged**